tools/lint-docstring

```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/env python3
+
+import argparse
+import importlib
+import pathlib
+import sys
+from os import listdir
+from os.path import isfile, join
+from typing import Dict, cast
+
+
+INITIAL_INDEX = 7
+
+# absolute path to zulip-terminal
+ROOT_DIRECTORY = pathlib.Path(__file__).resolve().parent.parent
+
+# new doc file has been created for the time being to compare it with the original
+DEVELOPER_DOC = join(ROOT_DIRECTORY, "docs", "developer-file-overview_new.md")
+
+COLUMN_WIDTHS = (23, 20, 103)
+SPACEBAR = " "
+
+# zulipterminal/scripts and zulipterminal/themes do not have any files
+# mentioned in the file-overview. Hence, their description has been
+# added separately.
+SCRIPTS = "zulipterminal/scripts"
+SCRIPTS_DESC = "Scripts bundled with the application"
+THEMES = "zulipterminal/themes"
+THEMES_DESC = "Themes bundled with the application"
+
+# leaving blank rows after exhausting files from each folder
+BLANK_LINE = f"| {23*SPACEBAR}| {20*SPACEBAR}| {103*SPACEBAR}|\n"
+
+
+def column_text_with_spaces(column_text, columnwise_widths=COLUMN_WIDTHS):
+    """
+    Returns the text parsed alongwith the adequate amount of spaces
+    after the text
+    """
+    folder_width, file_width, desc_width = columnwise_widths
+    folder_text, file_text, desc_text = column_text
+    folder_remaining_spaces = (folder_width - len(folder_text)) * SPACEBAR
+    file_remaining_spaces = (file_width - len(file_text)) * SPACEBAR
+    desc_remaining_spaces = (desc_width - len(desc_text)) * SPACEBAR
+    folder_name = folder_text + folder_remaining_spaces
+    file_name = file_text + file_remaining_spaces
+    file_desc = desc_text + desc_remaining_spaces
+    return (folder_name, file_name, file_desc)
+
+
+def write_to_file(list_of_new_files_to_add, current_doc_data, write_location) -> int:
+    """
+    Writes the list of rows containing folder name, file name(s) and their respective
+    descriptions to the developer-file-overview_new.md
+    """
+    current_doc_data[(write_location) : (write_location - 1)] = list_of_new_files_to_add
+    write_location = write_location + len(list_of_new_files_to_add) + 1
+    # write_location += 1
+    updated_data = "".join(current_doc_data)
+    with open(DEVELOPER_DOC, "w") as file:
+        file.write(updated_data)
+    return write_location
+
+
+def get_docstrings() -> Dict[str, Dict[str, str]]:
+    """
+    Returns a dictionary containing folder name which in turn
+    is a dictionary containing files and their respective descriptons
+    """
+    folder_paths = {
+        "zulipterminal": join(ROOT_DIRECTORY, "zulipterminal"),
+        "zulipterminal/cli": join(ROOT_DIRECTORY, "zulipterminal", "cli"),
+        "zulipterminal/config": join(ROOT_DIRECTORY, "zulipterminal", "config"),
+        "zulipterminal/ui_tools": join(ROOT_DIRECTORY, "zulipterminal", "ui_tools"),
+    }
+    total_files = {}
```

Comment:
total_files is really the input to this function; you might make it a constant, perhaps, if we don't calculate it dynamically at the moment.

To put it a different way this 'get' or 'extract' of the docstrings is separate from the calculation of which paths to extract them from.